consensus/consensus.go

```diff
@@ -113,6 +122,34 @@ type Consensus struct {
 
 	// List of offline Peers
 	OfflinePeerList []p2p.Peer
+
+	// Staking information finder
+	stakeInfoFinder StakeInfoFinder
+}
+
+// StakeInfoFinder returns the stake information finder instance this
+// consensus uses, e.g. for block reward distribution.
+func (consensus *Consensus) StakeInfoFinder() StakeInfoFinder {
+	return consensus.stakeInfoFinder
+}
+
+// SetStakeInfoFinder sets the stake information finder instance this
+// consensus uses, e.g. for block reward distribution.
+func (consensus *Consensus) SetStakeInfoFinder(stakeInfoFinder StakeInfoFinder) {
+	consensus.stakeInfoFinder = stakeInfoFinder
+}
+
+// StakeInfoFinder finds the staking account for the given consensus key.
```

Comment:
good interface design for reverse reference. :-D 👍

Reply:
<3